clang/test/dpct/query_api_mapping/cuRAND/test.cu

```diff
@@ -0,0 +1,651 @@
+// CURANDSETPSEUDORANDOMGENERATORSEED: CUDA API:
+// CURANDSETPSEUDORANDOMGENERATORSEED-NEXT:   curandGenerator_t g;
+// CURANDSETPSEUDORANDOMGENERATORSEED-NEXT:   curandSetPseudoRandomGeneratorSeed(g /*curandGenerator_t*/,
+// CURANDSETPSEUDORANDOMGENERATORSEED-NEXT:                                      ull /*unsigned long long*/);
+// CURANDSETPSEUDORANDOMGENERATORSEED-NEXT: Is migrated to:
+// CURANDSETPSEUDORANDOMGENERATORSEED-NEXT:   dpct::rng::host_rng_ptr g;
+// CURANDSETPSEUDORANDOMGENERATORSEED-NEXT:   g->set_seed(ull);
+// CURANDSETPSEUDORANDOMGENERATORSEED-EMPTY:
+
+// RUN: dpct --cuda-include-path="%cuda-path/include" --query-api-mapping=curandSetQuasiRandomGeneratorDimensions | FileCheck %s -check-prefix=CURANDSETQUASIRANDOMGENERATORDIMENSIONS
+// CURANDSETQUASIRANDOMGENERATORDIMENSIONS: CUDA API:
+// CURANDSETQUASIRANDOMGENERATORDIMENSIONS-NEXT:   curandGenerator_t g;
+// CURANDSETQUASIRANDOMGENERATORDIMENSIONS-NEXT:   curandSetQuasiRandomGeneratorDimensions(g /*curandGenerator_t*/,
+// CURANDSETQUASIRANDOMGENERATORDIMENSIONS-NEXT:                                           u /*unsigned int*/);
+// CURANDSETQUASIRANDOMGENERATORDIMENSIONS-NEXT: Is migrated to:
+// CURANDSETQUASIRANDOMGENERATORDIMENSIONS-NEXT:   dpct::rng::host_rng_ptr g;
+// CURANDSETQUASIRANDOMGENERATORDIMENSIONS-NEXT:   g->set_dimensions(u);
+// CURANDSETQUASIRANDOMGENERATORDIMENSIONS-EMPTY:
+
+// RUN: dpct --cuda-include-path="%cuda-path/include" --query-api-mapping=curandSetStream | FileCheck %s -check-prefix=CURANDSETSTREAM
+// CURANDSETSTREAM: CUDA API:
+// CURANDSETSTREAM-NEXT:   curandGenerator_t g;
+// CURANDSETSTREAM-NEXT:   curandSetStream(g /*curandGenerator_t*/, s /*cudaStream_t*/);
+// CURANDSETSTREAM-NEXT: Is migrated to:
+// CURANDSETSTREAM-NEXT:   dpct::rng::host_rng_ptr g;
+// CURANDSETSTREAM-NEXT:   g->set_queue(s);
+// CURANDSETSTREAM-EMPTY:
+
+/// Device API
+
+// RUN: dpct --cuda-include-path="%cuda-path/include" --query-api-mapping=curand | FileCheck %s -check-prefix=CURAND
+// CURAND: CUDA API:
+// CURAND-NEXT:   curandStateMtgp32_t *ps1;
+// CURAND-NEXT:   curandStateScrambledSobol64_t *ps2;
+// CURAND-NEXT:   curandStateSobol64_t *ps3;
+// CURAND-NEXT:   curandStateScrambledSobol32_t *ps4;
+// CURAND-NEXT:   curandStateSobol32_t *ps5;
+// CURAND-NEXT:   curandStateMRG32k3a_t *ps6;
+// CURAND-NEXT:   curandStatePhilox4_32_10_t *ps7;
+// CURAND-NEXT:   curandStateXORWOW_t *ps8;
+// CURAND-NEXT:   curand(ps1 /*curandStateMtgp32_t **/);
+// CURAND-NEXT:   curand(ps2 /*curandStateScrambledSobol64_t **/);
+// CURAND-NEXT:   curand(ps3 /*curandStateSobol64_t **/);
+// CURAND-NEXT:   curand(ps4 /*curandStateScrambledSobol32_t **/);
+// CURAND-NEXT:   curand(ps5 /*curandStateSobol32_t **/);
+// CURAND-NEXT:   curand(ps6 /*curandStateMRG32k3a_t **/);
+// CURAND-NEXT:   curand(ps7 /*curandStatePhilox4_32_10_t **/);
+// CURAND-NEXT:   curand(ps8 /*curandStateXORWOW_t **/);
+// CURAND-NEXT: Is migrated to:
+// CURAND-NEXT:   curandStateMtgp32_t *ps1;
```

Comment:
L173~L177 is cuda code,  it is not correct.

Reply:
deleted